### PR TITLE
Check compatibility when upgrading modules

### LINKED ISSUE #648: [MoveOS] Bug when remove and add value for same table key
When [publish_modules](https://github.com/rooch-network/rooch/blob/e619c845f9b532430c65858ba2e0d2ff1c4b9e86/moveos/moveos-stdlib/moveos-stdlib/sources/account_storage.move#L171C24-L171C24) in Move, if the module exists in table, we `remove` old module first and then `add` new one which will create `Op::New` and `Op::Delete` respectively when converting table data into `StateChangeSet`.

But the `StateChangeSet` can only store only one `Op` for a key of a table, so one of `Op` above will be covered when finish the transaction. The upgrading will **fail** without any transaction error.

### LINKED ISSUE #660: [MoveOS] Error when upgrading modules
When publish modules through `rooch move publish`, error happens.

Step to reproduce:

1.  publish module:
```
rooch move publish -p examples/entry_function_arguments/ --sender-account 0xb235db4c1208e024a2a118fec487bcf988cf26affac40572b2f6bf26ba523bdb --named-addresses rooch_examples=0xb235db4c1208e024a2a118fec487bcf988cf26affac40572b2f6bf26ba523bdb
```
This command will run successfully.

2. publish module again:
```
rooch move publish -p examples/entry_function_arguments/ --sender-account 0xb235db4c1208e024a2a118fec487bcf988cf26affac40572b2f6bf26ba523bdb --named-addresses rooch_examples=0xb235db4c1208e024a2a118fec487bcf988cf26affac40572b2f6bf26ba523bdb
```
This command will return with error in client:
```
Transaction error: Transaction execution failed: ExecutionFailure { location: StrView(Script), function: 0, code_offset: 0 }
```
error in server:
```
 moveos::moveos: execute tx(0x27ff…0004) error: Err(VMError { major_status: RESOURCE_ALREADY_EXISTS, sub_status: None, message: None, exec_state: None, location: Undefined, indices: [], offsets: [] })
```

### PR DESCRIPTION
**Pull request by pause125**
Check compatibility when upgrading modules,  add some test cases for module upgrading.
Also resolve #660 and resolve #648

### CODE REVIEW

File: moveos/moveos-stdlib/moveos-stdlib/sources/account_storage.move [hunk 1]
```diff
@@ -175,20 +175,34 @@ module moveos_std::account_storage {
         let len = vector::length(&modules);
         let (module_names, module_names_with_init_fn) = move_module::verify_modules(&modules, account_address);
         
+        let upgrade_flag = false;
         while (i < len) {
             let name = vector::pop_back(&mut module_names);
             let m = vector::pop_back(&mut modules);   
 
             // The module already exists, which means we are upgrading the module
             // TODO: check upgrade compatibility
```
Comment (jolestar) [inline moveos/moveos-stdlib/moveos-stdlib/sources/account_storage.move:184]: "// TODO: check upgrade compatibility" can be removed

**Review by baichuan3 — APPROVED** [review]
LGTM

File: moveos/moveos-stdlib/moveos-stdlib/sources/account_storage.move [hunk 2]
```diff
@@ -175,20 +175,34 @@ module moveos_std::account_storage {
...
                 move_module::request_init_functions(module_names_with_init_fn, account_address);
             };
             table::add(&mut account_storage.modules, name, m);
             i = i + 1;
+        };
+        
+        // Use MoveModule as flag to indicate module upgrading in this tx.
+        // Used to setting vm.mark_loader_cache_as_invalid(), which announce to 
+        // the VM that the code loading cache should be considered outdated. 
+        // TODO: whether define a new struct for this flag?
```
Comment (jolestar) [inline moveos/moveos-stdlib/moveos-stdlib/sources/account_storage.move:200]: I think define a new struct is better.

**Review by jolestar — APPROVED** [review]

**Comment by jolestar** [comment]
I merge this PR first. @pause125, you can refactor later.

**Pull request merged**